cmd/main.go

```diff
@@ -1,9 +1,49 @@
 package main
 
 import (
-	"github.com/dsauerbrun/cragcast/api"
+	"log"
+
+	"github.com/dsauerbrun/cragcast/pkg/http"
+	"github.com/spf13/cobra"
 )
 
+var bindHost string
+var port int
+
+func init() {
+	serveCmd.Flags().StringVar(&bindHost, "bind-host", "127.0.0.1", "The address to bind to. To accept all connections, use 0.0.0.0.")
+	serveCmd.Flags().IntVar(&port, "port", api.DefaultPort, "The port to listen on.")
+
+	rootCmd.CompletionOptions.DisableDefaultCmd = true
+	rootCmd.AddCommand(serveCmd)
+}
+
+var serveCmd = &cobra.Command{
+	Use:   "serve",
+	Short: "Starts the server",
+	Run: func(cmd *cobra.Command, args []string) {
+		api.Start(api.StartOptions{
+			BindHost: bindHost,
+			Port:     port,
+		})
+	},
+}
+
+var rootCmd = &cobra.Command{
+	Use:   "cragcast",
+	Short: "Weather-related information for the world's climbing areas.",
+	Run: func(cmd *cobra.Command, args []string) {
+		err := cmd.Help()
+		// err should never occur displaying help text. If it does,
+		// panic is acceptable.
+		if err != nil {
+			panic(err)
+		}
+	},
+}
+
 func main() {
-	api.InstantiateRoutes()
+	if err := rootCmd.Execute(); err != nil {
```

Comment:
so whenever we start the application, we're going to run the function from line 35, right? Is that what this is doing?

Reply:
There will be several entry-points. What you see here is **only** to facilitate the commandline execution, namely:

```
$ cragcast

Weather-related information for the world's climbing areas.

Usage:
  cragcast [flags]
  cragcast [command]

Available Commands:
  help        Help about any command
  serve       Starts the server

Flags:
  -h, --help   help for cragcast

Use "cragcast [command] --help" for more information about a command.
```

But, if you're starting the application from within another library (probably won't happen any time soon, you'd likely import our API (server) package and run that, or our WeatherClient if you're building your own server.

Let me know if that's not clear, I can throw together a simple diagram to describe the entrypoints.